src/lib/components/Cytoscape.react.js

```diff
@@ -0,0 +1,486 @@
+/**
+ * JavaScript Requirements: cytoscape
+ * React.js requirements: react-cytoscapejs
+ */
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import CytoscapeComponent from '../../react-cytoscapejs/src/component.js';
+import _ from 'lodash';
+
+
+export default class Cytoscape extends Component {
+    constructor(props) {
+        super(props);
+
+        this.handleCy = this.handleCy.bind(this);
+        this._handleCyCalled = false;
+    }
+
+    generateNode(event) {
+        const ele = event.target;
+
+        const isParent = ele.isParent();
+        const isChildless = ele.isChildless();
+        const isChild = ele.isChild();
+        const isOrphan = ele.isOrphan();
+        const renderedPosition = ele.renderedPosition();
+        const relativePosition = ele.relativePosition();
+        const parent = ele.parent();
+        const style = ele.style();
+        // Trim down the element objects to only the data contained
+        const edgesData = ele.connectedEdges().map(ele => {
+            return ele.data()
+        });
+        const childrenData = ele.children().map(ele => {
+            return ele.data()
+        });
+        const ancestorsData = ele.ancestors().map(ele => {
+            return ele.data()
+        });
+        const descendantsData = ele.descendants().map(ele => {
+            return ele.data()
+        });
+        const siblingsData = ele.siblings().map(ele => {
+            return ele.data()
+        });
+
+        const {timeStamp} = event;
+        const {
+            classes,
+            data,
+            grabbable,
+            group,
+            locked,
+            position,
+            selected,
+            selectable
+        } = ele.json();
+
+        let parentData;
+        if (parent) {
+            parentData = parent.data();
+        } else {
+            parentData = null;
+        }
```

Comment:
Why are we setting it to `null`?

Reply:
It's just a way of rewriting the code that you already have. Not a super big deal, but this way you have a "default" value :)

Reply:
@shammamah It makes sense! In this particular case i believe it doesn't affect how the app work since the subsequent if/else statement forces the variable to be assigned to a defined value. However I'd love to read more about the differences between null and undefined; feel free to share me an article/blog post about it :)

Reply:
Ok just added the relevant comments